aicsimageio/writers/ome_tiff_writer.py

```diff
@@ -205,7 +208,27 @@ def save(
             ] * num_images
         if channel_names is None or isinstance(channel_names[0], str):
             channel_names = [channel_names] * num_images  # type: ignore
-        if channel_colors is None or isinstance(channel_colors[0], int):
+
+        if channel_colors is not None:
+            if all(
+                [
+                    (
+                        channel_colors[img_idx] is None
+                        or isinstance(channel_colors[img_idx], list)
+                    )
+                    for img_idx in range(num_images)
+                ]
+            ):
+                single_image_channel_colors_provided = False
+            else:
+                single_image_channel_colors_provided = True
+
+            if channel_colors[0] is not None:
+                if isinstance(channel_colors[0], list):
+                    if isinstance(channel_colors[0][0], int):
+                        single_image_channel_colors_provided = True
```

Comment:
Since python short-circuits these checks, I think this can all be one `if` block with multiple statements `and`ed together (in order of the hierarchy defined here)